authn/keychain.go

```diff
@@ -41,6 +41,9 @@ func configDir() (string, error) {
 	if dc := os.Getenv("DOCKER_CONFIG"); dc != "" {
 		return dc, nil
 	}
+	if h := os.Getenv("HOME"); h != "" {
```

Comment:
I think that the correct fix would be to remove the `user.Current().homedir` and port this code (in addition to supporting $DOCKER_CONFIG):
https://github.com/google/containerregistry/blob/master/client/docker_creds_.py#L211

The python version more closely duplicates the actual behavior of the Docker client, which uses `%USERPROFILE%` on windows and `~` otherwise. The Docker client code is kinda spaghetti, and I don't remember where it is off the top of my head, but you can trust that this is a detail that has been burned into my brain :p

Reply:
Actually, it looks like the python lib has a bug here. DOCKER_CONFIG is supposed to be a directory, rather than the path including `config.json`

Reply:
I don't know anything about Windows, but I found https://stackoverflow.com/questions/7922270/obtain-users-home-directory which seems useful.

Added tests, some of which are `t.Skip`ped on non-Windows, but should work (famous last words).

OTOH, I can remove the `if runtime.GOOS == "windows"` check and just allow this weird behavior on non-Windows.

WDYT?

Reply:
I couldn't find the actual code, but the docs mention %USERPROFILE% and $HOME:
https://github.com/docker/cli/blob/2f3b13742fb1de2eb3b89ed2350182b8e83cf9dc/docs/reference/commandline/login.md#privileged-user-requirement